hack/verify-golint.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+#source "${KUBE_ROOT}/hack/lib/init.sh"
+
+#kube::golang::verify_go_version
+
+if ! which golint > /dev/null; then
+  echo 'Can not find golint, install with:'
+  echo 'go get -u golang.org/x/lint/golint'
+  exit 1
+fi
+
+cd "${KUBE_ROOT}"
+
+array_contains () {
+    local seeking=$1; shift # shift will iterate through the array
+    local in=1 # in holds the exit status for the function
+    for element; do
+        if [[ "$element" == "$seeking" ]]; then
+            in=0 # set in to 0 since we found it
+            break
+        fi
+    done
+    return $in
+}
+
+# Check that the file is in alphabetical order
+failure_file="${KUBE_ROOT}/hack/.golint_failures"
+if ! diff -u "${failure_file}" <(LC_ALL=C sort "${failure_file}"); then
+	{
+		echo
+		echo "hack/.golint_failures is not in alphabetical order. Please sort it:"
+		echo
+		echo "  LC_ALL=C sort -o hack/.golint_failures hack/.golint_failures"
+		echo
+	} >&2
+	false
+fi
+
+export IFS=$'\n'
+# NOTE: when "go list -e ./..." is run within GOPATH, it turns the k8s.io/kubernetes
+# as the prefix, however if we run it outside it returns the full path of the file
+# with a leading underscore. We'll need to support both scenarios for all_packages.
+all_packages=(
+	$(go list -e ./... | egrep -v "/(third_party|vendor|staging/src/k8s.io/client-go/pkg|generated|clientset_generated)" | sed -e 's|^k8s.io/kubernetes/||' -e "s|^_${KUBE_ROOT}/\?||")
```

Comment:
change `egrep` to `grep -E`, because egrep is non-standard and deprecated.

Reply:
try this `'$(go list -e ./... | egrep -v "/(third_party|vendor|staging/src/k8s.io/client-go/pkg|generated|clientset_generated)" | sed -e 's|^k8s.io/kubernetes/||' -e "s|^_${KUBE_ROOT}/\?||")'`